### PR TITLE
Implement user-definable keys

### PR DESCRIPTION
**Pull request by donnierussellii**
On run, if it doesn't already exist, a default "keybinds.txt" file will be created next to "prefs.txt" (see Prefs.h).

Keybinds file can be edited to bind keys to hotkey, fire or move actions. Key bindings take effect on next run.

movekeys.c was rewritten and is now much clearer.

If for some reason keybinds file cannot be read, default keys will still be properly set.

Also added/fixed:

Use SDL_TEXTINPUT events for printable char keys
Fix ALT key not working
Fix save hotkey
Quit hotkey was quitting without confirmation (!)
Fix randomly dropped modified keys (aka insidious evil)

Commits done.

### CODE REVIEW

**Comment by Interrupt** [comment]
You are a beast, man. Or a beastman. Either way, you should drop by our Discord channel sometime!
https://discord.gg/m45xPan

File: src/GameSrc/input.c [hunk 1]
```diff
@@ -1314,167 +1315,168 @@ void init_input(void) {
...
-       hotkey_add(DOWN('>'), DEMO_CONTEXT|EDIT_CONTEXT, stupid_slew_func, (void *)14);
-       hotkey_add(DOWN('<'), DEMO_CONTEXT|EDIT_CONTEXT, stupid_slew_func, (void *)15);
-
-       // 3d zoomin
-       hotkey_add(CONTROL('['), DEMO_CONTEXT|EDIT_CONTEXT, zoom_3d_func, (void *)TRUE);
-       hotkey_add(CONTROL(']'), DEMO_CONTEXT|EDIT_CONTEXT, zoom_3d_func, (void *)FALSE);
-
-       ckpoint_input("hotkeys in");
-
+    //
```
Comment (Interrupt) [inline src/GameSrc/input.c]: Why are these `//` instead of the `/*` block, is this coming from our code styling config?

Comment (donnierussellii) [reply src/GameSrc/input.c]: In order to /* */ the entire function, I converted those comment blocks  within the function into //
I wasn't sure if you could nest them that way.

Comment (Interrupt) [reply src/GameSrc/input.c]: Ah, I didn't see that the whole function was commented out. Makes sense now.

**Pull request merged**